airflow/providers/amazon/aws/sensors/sagemaker_training.py

```diff
@@ -72,6 +75,15 @@ def get_sagemaker_response(self):
         if self.print_log:
             if not self.log_resource_inited:
                 self.init_log_resource(self.get_hook())
+            if (
+                self.instance_count is None
+                or self.state is None
+                or self.last_description is None
+                or self.last_describe_job_call is None
+            ):
+                raise AirflowException(
```

Comment:
```suggestion
                raise ValueError(
```

is more appropriate than an AirflowException.